x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/use_hidden_by_flyout.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import type { RisonValue } from '@kbn/rison';
+import type { AlertsCasesTourSteps } from './tour_config';
+import {
+  hiddenWhenCaseFlyoutExpanded,
+  hiddenWhenLeftExpandableFlyoutExpanded,
+  SecurityStepId,
+} from './tour_config';
+import { useKibana } from '../../lib/kibana';
+import { URL_PARAM_KEY } from '../../hooks/use_url_state';
+import { getObjectFromQueryString } from '../../utils/global_query_string/helpers';
+
+interface UseHiddenByFlyoutProps {
+  tourId: SecurityStepId;
+  step: AlertsCasesTourSteps;
+}
+
+export const useHiddenByFlyout = ({ tourId, step }: UseHiddenByFlyoutProps) => {
```

Comment:
Yes, it would also be more descriptive if we re-name it to `useIsTourHiddenByFlyout`, The _is_ is important to quickly understand it returns a boolean.